src/byte_offsets.c

```diff
@@ -103,25 +103,40 @@ int RSByteOffset_Iterate(const RSByteOffsets *offsets, uint32_t fieldId,
   iter->buf.data = (char *) offsets_data;
   iter->buf.offset = offsets_len;
   iter->rdr = NewBufferReader(&iter->buf);
-  iter->curPos = 1;
   iter->endPos = offField->lastTokPos;
 
   iter->lastValue = 0;
 
-  while (iter->curPos < offField->firstTokPos && !BufferReader_AtEnd(&iter->rdr)) {
+  for (iter->curPos = 1; iter->curPos < offField->firstTokPos && !BufferReader_AtEnd(&iter->rdr); ++iter->curPos) {
     iter->lastValue = ReadVarint(&iter->rdr) + iter->lastValue;
-    iter->curPos++;
   }
 
-  iter->curPos--;
+  // If we reached the end of the stream before we reached the first token position, return an error
+  if (iter->curPos < offField->firstTokPos) {
+    return REDISMODULE_ERR;
+  }
+
+  // if range is [1, 1] we want curPos to be 0 so RSByteOffsetIterator_Next will return the first value
+  --iter->curPos;
```

Comment:
why don't we put it explicitly at 0 instead of -1 then? Not sure I see the relationship of the code with the comment.